### PR TITLE
AddWhere

### LINKED ISSUE #9: Create Where<T>
https://docs.microsoft.com/en-us/dotnet/api/system.linq.enumerable.where?view=netcore-3.1#System_Linq_Enumerable_Where__1_System_Collections_Generic_IEnumerable___0__System_Func___0_System_Boolean__

Reference existing code from [ShittyLinq.js.Where](https://github.com/ShittyLabs/ShittyLINQ.js/blob/develop/src/Where.js)

### PR DESCRIPTION
**Pull request by Jarrod2112**
ShittyLabs Where function and test.

### CODE REVIEW

File: test/Where.spec.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import { expect } from "chai";
+import { argv0 } from "process";
+import { DangerLinq } from "../src";
+import { DangerLinqFunction } from "../src/DangerLinqFunction";
+
+describe("DangerLinqFunction.Map", () => {
+    it("Should be a function", async () => {
+      const Where = await DangerLinq(DangerLinqFunction.Map);
```
Comment (the-pat) [inline test/Where.spec.ts]: To be consistent with the other test files, please use a lowercase `where` for the variable name.

Comment (the-pat) [inline test/Where.spec.ts]: This is the `Map` function, not the `Where` function.

File: test/Where.spec.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+      expect(Where).to.be.a("function");
+    });
+
+describe('Array#prototype#Where', function() {
+  const input = [1, 2, 3, 4, 5, 6];
+
+  it('Should return only even numbers', function() {
+    const expected = [2, 4, 6];
+    const actual = input.Where(x => x % 2 === 0);
```
Comment (the-pat) [inline test/Where.spec.ts]: DangerLinq doesn't add the function to the array prototype, the code won't run like this (unlike https://github.com/ShittyLabs/ShittyLINQ.js). The test should look something like:
```javascript
const Where = await DangerLinq(DangerLinqFunction.Where);
const actual = where(input, x => x % 2 === 0);

expect(actual).to.eql(expected);
```

**Review by the-pat — DISMISSED** [review]
@Jarrod2112, it looks like your tests are failing here.

Take a look at https://github.com/ShittyLabs/DangerLinq/blob/main/test/map.spec.ts. Notice how each test calls `const map = await DangerLinq(DangerLinqFunction.Map);`? I think that might be part of the reason the tests are failing.

**Comment by jreina** [comment]
@the-pat @Jarrod2112 adding `hacktoberfest-accepted` label since we're going to team up on this.

**Review by jreina — APPROVED** [review]

**Comment by jreina** [comment]
Thank you for your contribution. Happy hacking 🎃

**Pull request merged**